library/std/src/sync/mpmc/list.rs

```diff
@@ -213,6 +213,8 @@ impl<T> Channel<T> {
                     .compare_exchange(block, new, Ordering::Release, Ordering::Relaxed)
                     .is_ok()
                 {
+                    #[cfg(miri)]
+                    crate::thread::yield_now();
```

Comment:
The problem with this is that this will then always be present when Miri executes this code, even outside the test. Ideally we'd only enable this on our CI. This will require some new flag though...

Also there should be a comment explaining the point of this.

Reply:
Added a comment explaining why this point is interesting and a pointer to the test that uses it

Reply:
Thanks!

I guess now the main question is whether it's worth leaving in this `yield` for all Miri users or whether it should be somehow only applied for the test suite. It probably doesn't harm to always have it...

Reply:
I would also lean towards leaving there. It does yield at a very interesting point of the channel in general so having it there might catch future regressions sooner